pkg/roachprod/roachprod.go

```diff
@@ -435,7 +435,10 @@ func SQL(
 	if err != nil {
 		return err
 	}
-	return c.SQL(ctx, l, tenantName, cmdArray)
+	if len(cmdArray) == 0 {
+		return c.OpenSQLShell(ctx, l, tenantName)
+	}
+	return c.RunSQL(ctx, l, cmdArray)
```

Comment:
I'm boolean impaired today so perhaps I have just confused myself.  But does this end up minorly changing the behaviour of something like: `roachprod sql local:1 -- -u foo` where `foo` is a user in the cluster.  Previously this would open an interactive shell to the node. Now, I don't think it does.

Reply:
huh, you're right. The behavior on master is: if one node is specified, ssh onto that node, and pass the cmd to ./cockroach sql. If many nodes are specified, try parallel execution. It's a confusing API, and I think I have over simplified it. I'll change this PR to retain the behavior, and will let test-eng ponder if this API behavior should change.